os/Xtranssock.c

```diff
@@ -443,16 +442,17 @@ static XtransConnInfo _XSERVTransSocketReopen (
       return NULL;
     }
 
-    portlen = strlen(port) + 1; // include space for trailing null
+    size_t portnamelen = strlen(port) + 1;
+    size_t portlen = portnamelen;
 #ifdef SOCK_MAXADDRLEN
-    if (portlen < 0 || portlen > (SOCK_MAXADDRLEN + 2)) {
+    if (portlen > (SOCK_MAXADDRLEN + 2)) {
       prmsg (1, "SocketReopen: invalid portlen %d\n", portlen);
       return NULL;
     }
     if (portlen < 14) portlen = 14;
 #else
```

Comment:
I wonder why they don't have a `if (portlen < 14) portlen = 14;` line here in the `#else` branch for `#ifdef SOCK_MAXADDRLEN`? I guess it's just `SOCK_MAXADDRLEN` is always defined and that branch is never executed.

Reply:
maybe you'd like to check whether this actually can happen at all ?